.github/workflows/pull-request.yml

```
@@ -0,0 +1,451 @@
+        run: |
+          chmod +x scripts/githooks/check-localstorage-usage.js
+          node scripts/githooks/check-localstorage-usage.js --scan-entire-repo
+
+      - name: Compare translation files
+        run: |
+          chmod +x .github/workflows/scripts/compare_translations.py
+          python .github/workflows/scripts/compare_translations.py --directory public/locales
+
+      - name: Check if the source and target branches are different
+        if: ${{ github.event.pull_request.base.ref == github.event.pull_request.head.ref }}
+        run: |
```

actionlint (1.7.4)

"github.event.pull_request.head.ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions for more details

(expression)